addons/HotelPvP/scripts/Game/AFM_HotelPvP/Components/AFM_SpectatorComponent.c

```diff
@@ -0,0 +1,43 @@
+class AFM_SpectatorComponentClass : ScriptComponentClass
+{
+}
+
+class AFM_SpectatorComponent : ScriptComponent
+{
+	[RplProp()]
```

Comment:
As far as I understand the current code, `SetState` should only ever be executed on Server/Authority. I feel like simplifying the setter and using the Rpl `onRplName` feature to show the hint on the client would make it easier to follow what's going on.
```suggestion
	[RplProp(onRplName: "OnIsSpectatorUpdated", condition: RplCondition.OwnerOnly)]
```